app/models/driver.rb

```diff
@@ -0,0 +1,31 @@
+class Driver < ApplicationRecord
+  validates :name, presence: { message: "Please enter a name" }
+  validates :vin, presence:  {message: "Please enter a VIN" }
+
+  has_many :trips
+
+  def total_amount
+    total = 0
+
+    trips.each do |trip|
+      total += trip.cost
+    end
+    total -= (total *0.15)
+    return "$#{(total/100).round(2)}"
+  end
+
+  def average_rating
+    total = 0
+
+    trips.each do |trip|
+      total += trip.rating
+    end
+
+    if trips.length == 0
+      return "No trips yet"
```

Comment:
Note that it is fine to return two different data types from a method, but it is good practice to try to figure out how to return similar data rather than an error message. Instead of retuning this string you could return `nil` or `0` as a starting value.